### ISSUE TITLE
Downloadable voices or higher quality variants do not show up in Safari

### ISSUE THREAD

**Issue by HadrienGardeur** [HadrienGardeur/web-speech-recommended-voices#19]
On macOS, iOS and iPadOS, [Apple provides a pretty nice range of voices](https://github.com/HadrienGardeur/web-speech-recommended-voices?tab=readme-ov-file#macos) across various languages and regions.

Most of them require to go in system settings to download them, or a higher quality variant.

While this work well on desktop when running Chrome/Edge/Firefox, none of them show up when listing voices in Safari. This makes things even worse on iOS and iPadOS where Safari is the only option available, and all other browsers are built on top of it.

**Links:**

- TBD

**Comment by HadrienGardeur** [comment]
This bug is even worse than I expected when testing on recent builds of macOS/iOS:

- installing multiple variants of a voice results in these voices disappearing from Safari completely when calling `getVoices()`
- this can result in a language disappearing completely from Safari if all default voices have higher quality variants installed
- in addition, with the updated component for managing voices on macOS in macOS 15 Sequoia (which seems to be shared with iOS now), it's no longer possible to uninstall voice packs

This results in a fairly catastrophic situation where voices or even languages can disappear completely from Safari on macOS, without any way of going back.

I think we're reaching a point where this issue needs to be escalated through various organizations.

**Comment by jashson** [comment]
With the latest macOS Sequoia, iPadOS 18 and iOS18, the only high quality voice for german "Anna" has gone. Now there are only the Eloquence voices left which are totally unusable.

At least for many other languages like english, french, italian, ... there a still good quality voices available in Safari.

**Comment by david-athena** [comment]
We're running into this unfortunate issue currently. Though I do think it is possible to delete voices in Sequoia, see here: https://support.apple.com/en-sa/guide/mac-help/mchlp2290/mac

"Use your mouse or trackpad to swipe left on the voice you want to delete, then click Delete."

**Comment by HadrienGardeur** [comment]
> So it seems on Safari we're stuck with Samantha levels of quality.

Yeah and that's a real shame. macOS and iOS also have very high quality voices available with Siri, but they're not making them available through the Web Speech API or their native API.

As we expand our work on Readium Speech, we'll most likely introduce a concept of adapters, to connect to additional third-party APIs but I really hope that Apple will eventually wake up, this is crucial for accessibility.

**Comment by jashson** [comment]
With one of the latest macOS, iOS and iPadOS Updates the useable german voice "Anna" got back and is available again. Same to some additional english voices (Daniel for example got available again). Maybe just update to the latest Apple OS may fix the issue of unavailable quality voices.

**Comment by HadrienGardeur** [comment]
Good to know, thanks @jashson I'll test it out again and see how it behaves this time.

**Comment by HadrienGardeur** [comment]
I've tried the latest beta update for macOS and voices no longer disappear when installing a higher quality variant but:

- higher quality or downloadable voices remain unavailable
- Daniel shows up as both an English voice (UK) and French voice for me
- the French version of Daniel doesn't work (I'm not surprised, shouldn't be listed in the first place)

It's not as bad as before, but not much of an improvement either.

**Comment by gheorghiuradu** [comment]
Hopefully this will be fixed in macOS 26

**Comment by HadrienGardeur** [comment]
I will test iOS/iPadOS/macOS 26 as soon as they release the first public beta. I'm not targeting the developer beta because it's usually unstable and too removed from the final product.
